receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -388,23 +390,59 @@ func addNumberDatapoints(datapoints pmetric.NumberDataPointSlice, ls labels.Labe
 		dp.SetDoubleValue(sample.Value)
 
 		attributes := dp.Attributes()
-		for _, l := range ls {
-			if l.Name == "instance" || l.Name == "job" || // Become resource attributes
-				l.Name == labels.MetricName || // Becomes metric name
-				l.Name == "otel_scope_name" || l.Name == "otel_scope_version" { // Becomes scope name and version
-				continue
-			}
-			attributes.PutStr(l.Name, l.Value)
-		}
+		extractAttributes(ls).CopyTo(attributes)
 	}
 }
 
 func addSummaryDatapoints(_ pmetric.SummaryDataPointSlice, _ labels.Labels, _ writev2.TimeSeries) {
 	// TODO: Implement this function
 }
 
-func addHistogramDatapoints(_ pmetric.HistogramDataPointSlice, _ labels.Labels, _ writev2.TimeSeries) {
-	// TODO: Implement this function
+func addHistogramDatapoints(datapoints pmetric.ExponentialHistogramDataPointSlice, ls labels.Labels, ts writev2.TimeSeries) {
```

Comment:
You can probably reuse https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c56251822dcb61aeef7280358d7f4e7167cd99fa/receiver/prometheusreceiver/internal/metricfamily.go#L159 that I wrote for the prometheus receiver.

Preferably we put it in a common package.

Reply:
I took a look into this function, and this is using the metricGroup/metricFamily structs, with that, we will need to change a little of the code. Are your suggestion to use this code as reference, or to migrate it to a common package and use on both sides: receiver/prometheusreceiver and receiver/prometheuswritereceiver ?

Reply:
> Are your suggestion to use this code as reference, or to migrate it to a common package

Whichever works faster for you . We can refactor later to extract the common bits into a package.

Reply:
> We can refactor later to extract the common bits into a package.

Where should this package be placed?